### DISCUSSION TITLE
Map hillshading

### DISCUSSION THREAD

**Discussion by ramondevic** [devemux86/cruiser#112] [Q&A] [answered]
In Cruiser I can't find the "Map Hillshading" option that I used in K2, can you help me find it or is it not there?

**Cruiser**
![Screenshot_20230401083515](https://user-images.githubusercontent.com/122691401/229272939-b4b1c2c2-f95c-41f6-ad69-b506153aa084.png)

**K2**
![Screenshot_20230401083408](https://user-images.githubusercontent.com/122691401/229272954-9e91c8fe-3dda-4e61-954f-bfb20fe78639.png)

Thanks !

**Comment by devemux86** [comment]
Hi @ramondevic,

Cruiser had map hillshading in the past, but the server has been closed.
Hillshading is temporarily disabled as no servers appear to be available?

See also the discussion here:
- https://github.com/devemux86/cruiser/discussions/71

---

K2 hillshading was a paid service to [Mapilion](https://mapilion.com/) (owned by Kurviger).

**Comment by ramondevic** [comment]
Ok, I didn't know it was there and I can't find it.
We hope that it can be reactivated later.
Thanks !

**Comment by Ciutnik** [comment]
Hi. Perhaps an alternative could be an overlay with contour lines? When I was using Garmin 276c I created such a transparent overlay (based on free DEM data) and I loved it. The map became quasi-3D without losing readability in my opinion. But I realize that in our case it may be difficult or even impossible due to limited hardware resources.

**Reply by devemux86** [reply]
@Ciutnik 

> Perhaps an alternative could be an overlay with contour lines?

Contour lines are already available, if you activate offline maps + themes that contain them:
(in the map download manager)
- OpenAndroMaps
- Freizeitkarte

<img src="https://user-images.githubusercontent.com/3484020/229276039-04f29252-c615-4923-84b4-c78f867149f0.jpg" width="50%"/>

**Reply by Ciutnik** [reply]
I like Freizeitkarte maps, but I guess I'd prefer Mapsforge with a Biker theme (this is the best map readability for me) and contour lines as an optional overlay :)

**Reply by devemux86** [reply]
Contour lines must exist in both:
- Maps
- Themes

If you use the Biker theme with OpenAndroMaps or Freizeitkarte maps (which contain contour lines),
I could add the contour lines layer to the Biker theme?

Adding contour lines to offline maps is a complex process,
so I don't know when it will be possible to offer it in Mapsforge maps.

**Reply by Ciutnik** [reply]
> I could add the contour lines layer to the Biker theme?

I think there is little interest from users in this solution...

The density of the contour lines drawn may also be a problem. I know from experience that it looks different in different mountains. In my map, the contour lines were quite dense, because for the needs of the low and vast Carpathians. In the Alps, they would probably obscure the most important picture of the road map.

Edit: I found a picture of my GPS where you can see contour lines drawn every 25 m.

![11-02-17_15-21-42~3](https://user-images.githubusercontent.com/122647782/229280658-97dc0b17-6a0a-471f-963a-7e125e1ec61d.jpg)

**Comment by ramondevic** [comment]
Hillshading is more than anything to have a visual idea of where to go or not to go while looking at the map, but if you can't, nothing happens, I usually always prepare the tracks from the computer choosing the path I prefer to take.

As for the level curves, it would be interesting if you are cycling or on foot, but on a motorbike I don't find them necessary and I also think they would hinder the visibility of the map.

Thanks.

**Reply by devemux86** [reply]
@ramondevic you are right, map visibility is also important.

I will keep looking for hillshading solutions! :slightly_smiling_face:

**Comment by devemux86** [comment]
Coming in next Cruiser:

<img src="https://user-images.githubusercontent.com/3484020/231208554-f2235896-e817-4685-a489-4fea6cccf98b.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
Cool

**Comment by ramondevic** [comment]
Nice !

**Comment by devemux86** [comment, marked as answer]
Hillshading is available again in [Cruiser 3.0.18](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Available in "Settings | Map | Layers | Hillshading" or search settings for "hill".

**Reply by DolomitesTom** [reply]
Esri light is super for me, changes only slightly the colors, looks very good.

**Reply by devemux86** [reply]
It is not fully transparent, probably more improvements can be made in the future. :slightly_smiling_face:

To save battery, it is recommended to turn it off in navigation:
- https://github.com/devemux86/cruiser/discussions/139

**Comment by DolomitesTom** [comment]
> It is not fully transparent, probably more improvements can be made in the future. 🙂

To my opinion it can't stay like it is, at least the Esri light. I like the effect of not being completely transparent.

Only the ESRI dark is a bit too dark.

**Reply by devemux86** [reply]
@DolomitesTom 

> To my opinion it can't stay like it is, at least the Esri light.

Do you mean the light version can stay like it is?

> Only the ESRI dark is a bit too dark.

We can try more post-processing on this version.

**Reply by DolomitesTom** [reply]
> Do you mean the light version can stay like it is?

Yes, looks nice. Don't know what the others think. They could comment too.

**Comment by airald** [comment]
Nice! Very useful to "see" the terrain.

**Comment by DolomitesTom** [comment]
After driving under several rainy, cloudy and sunny conditions I have to correct myself. Even if the ESRI light is good looking, it takes away some brightness and clearness of the colors and this is unfavourable. I guess it is better to improve the transparency or leave the ESRI light as it is and make the ESRI dark transparent and rename it into ESRI transparent.

**Reply by devemux86** [reply]
@DolomitesTom thanks for the testing!

The issue is that these hillshading tiles are not fully transparent (like the previous ones).
Changing their transparency affects the shading on the map: making shades less visible.

Note that only the map background is affected by the hillshading,
the icons and labels are rendered clearly on top.

However, we should really have a better visible solution. :slightly_smiling_face:
